modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -192,6 +196,11 @@ public class TxManagerImpl implements TxManager, NetworkMessageHandler {
 
     private final ReplicaService replicaService;
 
+    /** Registry of locally started active transactions. */
+    private final Map<UUID, InternalTransaction> transactions = new ConcurrentHashMap<>();
```

Comment:
This map has a risk for performance. I think we can reuse another existing storage (for example VolatileTxStateMetaStorage).

Reply:
@vldpyatkov , we need registry of running transactions to be able to abort pending transaction (command `KILL TRANSACTION TxId`).
`VolatileTxStateMetaStorage` contains only states for RW transactions.

Can you suggest how the command should be designed to abort any running transaction (including read-only transactions) without such mapping?

Reply:
After private discussions, the PR was reworked so that the system view now uses existing collections.